lib/xrp/XrpRpc.js

```diff
@@ -0,0 +1,248 @@
+const RippleAPI = require('ripple-lib').RippleAPI;
+const promptly = require('promptly');
+const util = require('util');
+const EventEmitter = require('events');
+
+const passwordPromptAsync = util.promisify(promptly.password);
+
+class XrpRpc {
+  constructor(config) {
+    this.config = config;
+    const {
+      rpcPort,
+      host,
+      protocol,
+      address
+    } = config;
+    const connectionString = `${protocol}://${host}:${rpcPort}`;
+    this.rpc = new RippleAPI({
+      server: connectionString
+    });
+    this.address = address;
+    this.emitter = new EventEmitter();
+  }
+
+  async asyncCall(method, args) {
+    return new Promise((resolve) => {
```

Comment:
Why is this one all nesting promises instead of awaiting them? 

If you want that structure, you could return `this.rpc.connect.then(...)`

Reply:
could add `alreadyConnected = false` as an optional param, and only do the connect disconnect logic when that's false, that way you can use asyncCall for all the methods and get your connect disconnect logic without duplicating those lines everywhere

Reply:
Doing that now

Reply:
Done